src/Uno.UI/UI/Xaml/Controls/Frame/Frame.cs

```diff
@@ -326,9 +326,9 @@ private bool InnerNavigate(PageStackEntry entry, NavigationMode mode)
 					null
 				);
 
+				Navigated?.Invoke(this, navigationEvent);
```

Comment:
Events most likely have to be moved before `Content = CurrentEntry.Instance;` above for `x:Bind` to done after  navigatedto properly.

Reply:
Unfortunately, it must stay after setting the `Content` - in UWP `Content` is already set when `Navigated` is raised. It seems the `Loading` and `Loaded` events need to be initiated by the frame when the page is inside of it.

```
Frame navigating
Page constructor
Frame.Content == EventOrdering.MainPage
Frame navigated
Page navigated to
Page loading
Page loaded
```

Reply:
@jeromelaban I have created #2895 for it. It seems it will be a very complex issue.